feat: Meltano Cloud schedule enable/disable command

WillDaSilva opened the pull request:
Tested in the AWS sandbox. Confirmed that it sets both the `enabled` value in the project schedules DynamoDB table, and actually enables/disables the EventBridge Rule that runs the schedule.

Closes #7303

edgarrmondragon committed:
feat: Add `meltano-cloud logs print`

WillDaSilva committed:
WIP: Add `meltano cloud schedule (enable|disable)` command

Closes #7303

WillDaSilva committed:
Address mypy issue

edgarrmondragon commented on src/cloud-cli/meltano/cloud/api/client/__init__.py:
```diff
@@ -60,6 +60,10 @@ def __init__(self, config: MeltanoCloudConfig | None = None) -> None:
         self.runner_api_url = self.config.runner_api_url
         self.api_key = self.config.runner_api_key
         self.runner_secret = self.config.runner_secret
+        try:
+            self.version = version("meltano.cloud.cli")
+        except Exception:
+            self.version = "0.0.0-dev"
```
FWIW I found that `version("meltano.cloud.cli")` works when running from the `src/cloud-cli`, while `version("meltano")` works when running from the root of the meltano repo.

WillDaSilva committed:
Replace `client/__init__.py` with `client.py`

WillDaSilva committed:
Fetch Cloud CLI version from Meltano package as a fallback

WillDaSilva committed:
Update Cloud CLI version

This'll be handled like all other version numbers once it's merged into `main`.

edgarrmondragon committed:
Merge branch 'cloud' into feat/cli-get-execution-logs

WillDaSilva committed:
Merge branch 'cloud' into 7303-cloud-cli-schedule

WillDaSilva committed:
Merge branch 'feat/cli-get-execution-logs' into 7303-cloud-cli-schedule

edgarrmondragon committed:
Update src/cloud-cli/meltano/cloud/cli/logs.py

Co-authored-by: Will Da Silva <will@willdasilva.xyz>

WillDaSilva committed:
Merge branch 'feat/cli-get-execution-logs' into 7303-cloud-cli-schedule

edgarrmondragon committed:
fix: Restore `MeltanoCloudConfig.find` (#7396)

* fix: Restore MeltanoCloudConfig.find

* Avoid changing user's config

WillDaSilva committed:
Merge branch 'feat/cli-get-execution-logs' into 7303-cloud-cli-schedule

WillDaSilva committed:
Use new Cloud CLI config config attributes

WillDaSilva committed:
Improve error message when schedule or deployment is not found

WillDaSilva commented:
I've re-targeted this PR to `feat/cli-get-execution-logs` because it has the config fix from https://github.com/meltano/meltano/pull/7396

https://github.com/meltano/meltano/pull/7379 should be merged into `cloud` before this PR is merged.

WillDaSilva committed:
Address lint issues

WillDaSilva committed:
Merge branch 'cloud' into feat/cli-get-execution-logs

WillDaSilva committed:
Address mypy issue

WillDaSilva committed:
Merge branch 'feat/cli-get-execution-logs' into 7303-cloud-cli-schedule

edgarrmondragon committed:
Add docs for logs command

edgarrmondragon committed:
Merge branch 'cloud' into feat/cli-get-execution-logs

edgarrmondragon committed:
Add logs CLI tests

edgarrmondragon committed:
Apply suggestions from code review

Co-authored-by: Will Da Silva <will@willdasilva.xyz>

edgarrmondragon committed:
Move CLI fixtures to `conftest.py`

WillDaSilva committed:
Merge branch 'feat/cli-get-execution-logs' into 7303-cloud-cli-schedule

WillDaSilva committed:
Address Pytest warnings

WillDaSilva committed:
Merge branch 'cloud' into 7303-cloud-cli-schedule

WillDaSilva committed:
Update Cloud CLI documentation

WillDaSilva committed:
Add tests for schedule CLI

WillDaSilva committed:
Address lint issues

edgarrmondragon reviewed (approved):
This is great!

WillDaSilva committed:
Merge branch 'cloud' into 7303-cloud-cli-schedule
